Source/API/CBLReplication.m

```diff
@@ -343,7 +343,7 @@ - (void) updateStatus: (CBLReplicationStatus)status
         return;
     if (status == kCBLReplicationStopped) {
         _started = NO;
-        [_database forgetReplication: self];
+        [self forgetReplication];
```

Comment:
We need to removing the observer here because the updateStatus method is actually called on the database thread. For example If I manually put some sleep as below, it will certainly crash.

In bg_updateProgress method:
```
__weak CBLReplication *weakSelf = self;
    [_database doAsync: ^{
        CBLReplication *strongSelf = weakSelf;
        [strongSelf updateStatus: status error: error processed: changes ofTotal: total
                          serverCert: serverCert];
        cfrelease(serverCert);
    }];
    
    if (status == kCBLReplicationStopped) {
        sleep(1);
        [self bg_setReplicator: nil];
    }
```

Another way to fix this would be moving if (status == kCBLReplicationStopped) block up before [_database doAsync: ]. What do you think?